### PR TITLE
Rails 3 Compatibility

### PR DESCRIPTION
**Pull request by zdennis**
# What does this PR do?

This PR updates `is_paranoid` to work with Rails 3 (and it drops Rails 2 support).

This also includes the work done in PR #9 and builds on them.

## Test Plan

Run the test suite locally:

```
bundle && bundle exec spec spec/is_paranoid_spec.rb
```

### CODE REVIEW

File: lib/is_paranoid.rb [hunk 1]
```diff
@@ -36,7 +57,11 @@ def is_paranoid opts = {}
     # and field_not_destroyed are). All exceptions require using
     # exclusive_scope (see self.delete_all, self.count_with_destroyed,
     # and self.find_with_destroyed defined in the module ClassMethods)
-    default_scope :conditions => {destroyed_field => field_not_destroyed}
+    default_scope do
+      unless IsParanoid.disabled?
```
Comment (a-warner) [inline lib/is_paranoid.rb]: is this what we want? I'd be worried about a case in development like:

```
# Song model not autoloaded yet
>> IsParanoid.disable { Song.first }
# Song model is now loaded without the right `deleted_at` default scope
>> Song.first
# potentially returns a deleted song
```

 also can `default_scope` take a block in rails 2?

Comment (zdennis) [reply lib/is_paranoid.rb]: Ooooh. Great question!

It looks like `default_scope` will do the right thing even if the model is autloaded from within a `IsParanoid.disable` block.

```bash
Loading development environment (Rails 3.2.22)
[1] dev (main)> Object.const_defined? :Song
=> false
[2] dev (main)> IsParanoid.disable { Song.first } ; true
  Song Load (8.3ms)  SELECT "songs".* FROM "songs" LIMIT 1
=> true
[3] dev (main)> Song.first ; true
  Song Load (1.8ms)  SELECT "songs".* FROM "songs" WHERE "songs"."deleted_at" IS NULL LIMIT 1
=> true 
```

> also can default_scope take a block in rails 2?

It does not look like it supports a block in Rails 2. Reference: https://github.com/rails/rails/blob/2-3-stable/activerecord/lib/active_record/base.rb#L2208-L2210

Comment (a-warner) [reply lib/is_paranoid.rb]: got it, thanks!

**Comment by zdennis** [comment]
I've rebased in `master`, resolved conflicts, and re-ran the test suite:

Screen Shot 2019-06-11 at 4 06 16 PM

**Comment by nightpool** [comment]
🎉 🎉

**Pull request merged**